_posts/2018-12-29-dask-on-hpc.md

```diff
@@ -0,0 +1,139 @@
+
+With tools like [Dask Jobqueue](https://jobqueue.dask.org) and [Dask MPI](https://mpi.dask.org)
+there is no need of any boilerplate bash code with `qsub/sbatch/...`.
+
+Dask interacts natively with our existing job schedulers (SLURM/SGE/SGE/LSF/...)
+so there is no additional system to set up and manage between users and IT.
+All the infrastructure that we need is already in place.
+
+Also, interactive analysis at scale and auto scaling is just awesome, and lets
+us use our existing infrastructure in new ways and improves our occupancy
+generally.
+
+
+### 3.  Aimed for scientific processing
+
+Dask is compatible with scientific data formats like HDF5, NetCDF, Parquet, and
+so on.  This is because Dask works with other libraries within the Python
+ecosystem, like XArray, which already has strong support for scientific data
+formats and processing.
+
+
+### 4.  Versatility of APIs
+
+And yet Dask is not designed for any particular workflow, and instead can
+provide infrastructure to cover a variety of different problems within an
+institution.  Everything is possible.
+
+
+### 5.  Versatility of Infrastructure
+
+Dask is compatible with laptops, servers, HPC systems, and even cloud
+computing if needed.  The environment can change with very little code
+adaptation.
+
+Dask is more than just a tool to us; it is a gateway to thinking about a
+completely different way of providing computing infrastructure to our users.
+Dask opens up the door to cloud computing technologies (such as elastic scaling
+and object storage) and makes us rethink what an HPC center should really look
+like!
+
+### 6.  Cost and Collaboration
+
+Dask is free and open source, which means we do not have to rebalance our
+budget and staff to address the new immediate need of data analysis tools.
+There are no licenses, and we have the ability to make changes when necessary.
+
+
+# What needs work
+
+### 1.  Heterogeneous resources handling
+
+Often we want to include different kinds of HPC nodes in the same deployment.
+This includes situations like the following:
+
+-  Workers with low or high memory
+-  Workers with GPUs
+-  Workers from different node pools
+
+Dask provides some support for this heterogeneity already, but not enough.
+We see two major opportunities for improvement.
+
+-  Tools like Dask-Jobqueue should make it easier to manage multiple worker
+   pools within the same cluster.  Currently the deployment solution assumes
+   homogeneity.
+-  It should be easier for users to specify which parts of a computation
+   require different hardware.  The solution today works, but requires more
+   detail from the user than is ideal.
+
+
+### 2.  Coarse-Grained Diagnostics
+
+Dask provides a number of profiling tools that give you diagnostics at the
+individual task-level, but there is no way today to analyze or profile your
+Dask application at a coarse-grained level.  Having more tools to analyze bulk
+performance would be helpful when making design decisions.
```

Comment:
One tangible idea here would be a benchmarking suite that helps users make decisions about how to use dask most effectively.